ui/e2e/src/pages/DashboardPage.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Page, Locator } from '@playwright/test';
+import { PanelEditor } from './PanelEditor';
+import { PanelGroup } from './PanelGroup';
+
+type PanelGroupConfig = {
+  name: string;
+};
+
+/**
+ * Perses App dashboard page.
+ */
+export class DashboardPage {
```

Comment:
The return of the page object! 🎉  I remember this pattern from my Selenium days, so kinda feels like a blast from the past a bit. I also remember there being some debate in the community about this pattern over the years, with Cypress in particular encouraging folks not to do it.

Is this encouraged in Playwright? I don't feel super strongly about it, just might be worth doing a little digging into pros/cons since I recall it being a bit of a 🌶️ topic. As someone who's clearly not an expert here, I could use the info too. Could make for a fun discussion topic.

Reply:
My understanding is that page objects are still commonly used in not-cypress use cases. My impression (although admittedly, I haven't dug into it a ton) is that cypress was pushing a different approach to similar problems (e.g. writing DRY test code, not tightly coupling tests to selectors) with their "app actions" pattern. A lot of the FUD around page objects seemed to come from cypress. 

As far as I can tell, Playwright doesn't promote an alternative pattern like that. They have an advanced feature called [fixtures](https://playwright.dev/docs/test-fixtures) that can do some similar things, but it seems more like a tool to use alongside page objects than instead of them. We could organize the selectors and helper functions using some other abstraction, but that feels a bit like reinventing the wheel.